src/rhash/hash.c

```diff
@@ -1143,6 +1143,76 @@ static int rc_hash_nintendo_ds(char hash[33], const char* path)
   return rc_hash_finalize(&md5, hash);
 }
 
+static uint32_t rc_read_bigendian_u32_value(void* file_handle)
+{
+  char le_buffer[4];
+  char be_buffer[4];
+  rc_file_read(file_handle, le_buffer, 4);
+  for (int ix = 0; ix < 4; ix++)
+  {
+    be_buffer[3 - ix] = le_buffer[ix];
+  }
+  return *((uint32_t*)be_buffer);
```

Comment:
I assume this is coming from some pattern in Dolphin, which is not endian safe (so in practice ever in the next 20 or so years this doesn't matter, only if Dolphin ever decides to become endian safe or some new emulator that supports a big endian machine pops up and supports rcheevos will this matter in practice). Not a reason to do this anyways, better to not rely on these assumptions.

Reply:
Gamecube and Wii are both big-endian while Dolphin, at least in the code I referenced to get the addresses and filesizes the hash algorithm needs, appears to be little-endian. That said, I agree with Jamiras, doing the math this way is probably preferable because it's unambiguous - though I'll still throw it in a function because I have to reuse it quite a few times.

Reply:
Compilers can optimize it for just a byteswap, but unfortunately that's not the case with MSVC: https://godbolt.org/z/5j76vTna5

On Windows you can use byteswap functions directly, if the function is called on tight loops it can help improve performace: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/byteswap-uint64-byteswap-ulong-byteswap-ushort?view=msvc-170

If you need it to be cross-platform, this can help: https://stackoverflow.com/questions/41770887/cross-platform-definition-of-byteswap-uint64-and-byteswap-ulong

Reply:
We'd need to take things a step farther and determine if a byteswap was needed in the first place. Using the explicit shifts ensure we translate the data from the ROM/ISO correctly, regardless of the endianness of the host system. 

And I'm not terribly worried about the performance for pulling a couple dozens integers out of the file.